lib/data/network/media/media_api.dart

```diff
@@ -34,4 +35,23 @@ class UploadFileAPI {
 
     return UploadFileResponse.fromJson(response);
   }
+
+  Future<UrlPreviewResponse> getUrlPreview({
+    required Uri uri,
+    int? ts,
+  }) async {
+    final response = await _client
+        .get(
+          HomeserverEndpoint.getPreviewUrlServicePath
+              .generateHomeserverIdentityEndpoint(),
```

Comment:
What? Why we need the identity endpoint here?

Reply:
it's same `upload`

Reply:
IMO, it is a mistake from the past, please use the exactly name for it